pypuppetdb/api/__init__.py

```diff
@@ -11,39 +11,45 @@
     EmptyResponseError,
     UnsupportedVersionError,
     APIError,
-    )
+)
 
 log = logging.getLogger(__name__)
 
 API_VERSIONS = {
-    2: 'v2',
-    3: 'v3',
+    4: 'v4',
 }
 
 ENDPOINTS = {
-    2: {
-        'facts': 'facts',
-        'fact-names': 'fact-names',
-        'nodes': 'nodes',
-        'resources': 'resources',
-        'metrics': 'metrics',
-        'mbean': 'metrics/mbean',
-    },
-    3: {
-        'facts': 'facts',
-        'fact-names': 'fact-names',
-        'nodes': 'nodes',
-        'resources': 'resources',
-        'catalogs': 'catalogs',
-        'metrics': 'metrics',
-        'mbean': 'metrics/mbean',
-        'reports': 'reports',
-        'events': 'events',
-        'event-counts': 'event-counts',
-        'aggregate-event-counts': 'aggregate-event-counts',
-        'server-time': 'server-time',
-        'version': 'version',
-    },
+    4: {
+        'facts': 'pdb/query/v4/facts',
+        'fact-names': 'pdb/query/v4/fact-names',
+        'nodes': 'pdb/query/v4/nodes',
+        'resources': 'pdb/query/v4/resources',
+        'catalogs': 'pdb/query/v4/catalogs',
+        'mbean': 'metrics/v1/mbeans',
+        'reports': 'pdb/query/v4/reports',
+        'events': 'pdb/query/v4/events',
+        'event-counts': 'pdb/query/v4/event-counts',
+        'aggregate-event-counts': 'pdb/query/v4/aggregate-event-counts',
+        'server-time': 'pdb/meta/v1/server-time',
+        'version': 'pdb/meta/v1/version',
+        'environments': 'pdb/query/v4/environments',
+        'factsets': 'pdb/query/v4/factsets',
+        'fact-paths': 'pdb/query/v4/fact-paths',
+        'fact-contents': 'pdb/query/v4/fact-contents',
+        'edges': 'pdb/query/v4/edges',
+    }
+}
+
+PARAMETERS = {
+    4: {
+        'order_by': 'order_by',
+        'include_total': 'include_total',
+        'count_by': 'count_by',
+        'counts_filter': 'counts_filter',
+        'summarize_by': 'summarize_by',
+        'server_time': 'server_time',
+    }
 }
```

Comment:
These maps were left over from a pull request that attempted to maintain backward compatibility. Since we no longer contain the functionality of previous Query API versions I agree with you about reducing these to a single-level map.